node_exporter.go

```diff
@@ -136,6 +136,22 @@ func (h *handler) innerHandler(filters ...string) (http.Handler, error) {
 	return handler, nil
 }
 
+func registerPprof(mux *http.ServeMux, enabled bool) {
+	if !enabled {
+		return
```

Comment:
If not enabled, maybe static page informing that `--pprof.enabled` is needed could be useful?

Reply:
You mean that when i try to request '/debug/pprof/' I should get 'Please, enable --debug.pprof in settings', not '404 page not found'?

Reply:
Yep, exactly! I was trying out this branch and did not set the command line flag `--pprof.enabled` and got redirected to node_exporters home page with no errors, not even a 404. 

Maybe something like `Debug pprof not enabled, to see profiling info set the command line flag "--pprof.enabled"`.
And yes, I meant `pprof.enabled` and not `debug.pprof` :stuck_out_tongue:. Looking at the PR, that's the flag you added:

https://github.com/prometheus/node_exporter/pull/1877/files#diff-7907251477470326df65250bf7fa699e0756ce5d6282e27b34c112132a7ef6b5R182

Reply:
That's just a suggestion though, I'm not a maintainer :slightly_smiling_face:

Reply:
It should be enabled by default, so there is no need for such a message I'd say. Also see: https://github.com/prometheus/node_exporter/pull/1877#issuecomment-739561044

Reply:
Okay, I just set default 'enabled'

The default behavior is 'default golang 404 page' now.